p2p/src/sync/peer/block_manager.rs

```diff
@@ -688,7 +692,12 @@ where
         {
             let new_block_headers = new_block_headers.clone();
             self.chainstate_handle
-                .call(move |c| Ok(c.preliminary_headers_check(&new_block_headers)?))
+                .call(move |c| {
+                    Ok(c.preliminary_headers_check(
+                        &new_block_headers,
+                        first_header_prev_block_index.block_height(),
```

Comment:
I think this isn't the correct way to do this. The right way would make the check depend on the block height, not on the string of blocks and their coincidental first block height.

Reply:
`preliminary_headers_check` only checks the first header in the list, so only one height is required. I made it more explicit though